.github/workflows/ios.yml

```diff
@@ -12,69 +12,129 @@ jobs:
+
       - name: Decode the RSA key for use with ssh
         env:
           FILE_DECRYPTION_PASSPHRASE: ${{ secrets.FILE_DECRYPTION_PASSPHRASE }}
         run: |
           gpg --quiet --batch --yes --decrypt --passphrase="$FILE_DECRYPTION_PASSPHRASE" --output ./ios/cfg/mapswipe.dev_at_gmail_rsa_key_for_travis_ci ./ios/cfg/mapswipe.dev_at_gmail_rsa_key_for_travis_ci.gpg
           chmod 600 ./ios/cfg/mapswipe.dev_at_gmail_rsa_key_for_travis_ci
+
+      - name: Decode the apple connect apicert
+        env:
+          FILE_DECRYPTION_PASSPHRASE: ${{ secrets.FILE_DECRYPTION_PASSPHRASE }}
+        run: |
+          gpg --quiet --batch --yes --decrypt --passphrase="$FILE_DECRYPTION_PASSPHRASE" --output ./ios/AuthKey_4B2GC5YX57.p8 ./ios/cfg/MapSwipe_apple_release_key_AuthKey_4B2GC5YX57.p8.gpg
+
       - name: Add gitlab to ssh_known_hosts
         env:
           SSH_KNOWN_HOSTS: ${{ secrets.SSH_KNOWN_HOSTS }}
         run: echo "$SSH_KNOWN_HOSTS" > ~/.ssh/known_hosts
+
       - name: Build the app and run unit tests
         # The test lane refreshes the development certificates, compiles the app and launches the unit tests.
         # Actual Tests have been disabled since we had issues launching the simulator, but the app is being compiled.
         # only build locally if not on a tagged commit, as the next fastlane runs will run the build anyway
         if: startsWith(github.ref, 'refs/heads/')
         env:
+          # select the version of Xcode we want
+          DEVELOPER_DIR: /Applications/Xcode_12.4.app/Contents/Developer
           MATCH_PASSWORD: ${{ secrets.MATCH_PASSWORD }}
         run: cd ios && bundle exec fastlane test
+
       - name: Release to testflight
         # only deploy a release on a non-beta tag
         # FIXME: there does not seem to be a way with github actions (as of may 2021) to get
         # both tag and branch, so for now we will rely on the user not tagging randomly
         if: startsWith(github.ref, 'refs/tags/') && endsWith(github.ref, '-beta')
         env:
+          DEVELOPER_DIR: /Applications/Xcode_12.4.app/Contents/Developer
           FASTLANE_PASSWORD: ${{ secrets.FASTLANE_PASSWORD }}
           MATCH_PASSWORD: ${{ secrets.MATCH_PASSWORD }}
         run: cd ios && bundle exec fastlane beta
+
       - name: Release to AppStore
         # only deploy a release on a beta tag
         if: startsWith(github.ref, 'refs/tags/') && !endsWith(github.ref, '-beta')
         env:
+          DEVELOPER_DIR: /Applications/Xcode_12.4.app/Contents/Developer
           FASTLANE_PASSWORD: ${{ secrets.FASTLANE_PASSWORD }}
           MATCH_PASSWORD: ${{ secrets.MATCH_PASSWORD }}
         run: cd ios && bundle exec fastlane release
+
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: build.log
+          path: ~/Library/Logs/gym/*.log
+
+      - name: Find output files
```

Comment:
do we still need this? looks like you tried to log which files there are

Reply:
Removed, good catch.